src/python/pants/backend/awslambda/python/aws_architecture.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+# NB: This file needs to be separate from target_types.py to avoid import cycles.
+
+from enum import Enum
+
+from pants.engine.target import StringField
+from pants.util.strutil import help_text
+
+
+class AWSLambdaArchitecture(str, Enum):
+    X86_64 = "x86_64"
+    ARM64 = "arm64"
```

Comment:
I originally had this in `faas.py` but it turns out that only AWS lets you choose the architecture for function execution; GCF [only supports x86](https://cloud.google.com/functions/docs/concepts/execution-environment#python).